Add Youtube documentation (#27378)
home-assistant/home-assistant.github.io: source/_integrations/youtube.markdown

---
title: YouTube
description: Instructions on how to integrate YouTube within Home Assistant.
ha_category:
  - Sensor
ha_release: '2023.6'
ha_iot_class: Cloud Polling
ha_config_flow: true
ha_codeowners:
  - '@joostlek'
ha_domain: youtube
ha_platforms:
  - sensor
ha_integration_type: service
---

The YouTube integration allows you to connect YouTube channels to Home Assistant.

## Prerequisites

You need to configure developer credentials to allow Home Assistant to access your Google Account.

> These credentials are the same as the ones for [Nest](/integrations/nest) and [Google Sheets](/integrations/google_sheets).

Reviewer: Let's also mention Google Mail. All 3 can use the same app credentials as Youtube. Maybe we also want to include a section for unblocking popups in case people might open an issue for not getting a window in the web browser. This sounds silly but it can happen and will happen as Home Assistant becomes more and more mainstream.

Author: How do you mean unblocking popups? If you mean the message you get when you get on those shady websites, I think home assistant just opens a separate tab. And if it was the case with all Application Credentials and it is an issue, we should template it imo

These are not the same as the one for [Google Calendar](/integrations/google).
If you have already set up the correct credentials, you can do step 1 and then skip to step 13 on the below instructions.

{% details "Generate Client ID and Client Secret" %}

This section explains how to generate a Client ID and Client Secret on
[Google Developers Console](https://console.cloud.google.com/apis/library/youtube.googleapis.com).

1. First, go to the Google Developers Console to enable the [YouTube Data API v3](https://console.cloud.google.com/apis/library/youtube.googleapis.com?project=home-assistant-17698)

> 2. The wizard will ask you to choose a project to manage your application. Select a project and click continue.

Reviewer 2: ```suggestion
2. The wizard will ask you to choose a project to manage your application. Select a project and select **Continue**.
```
I prefer _select_ over _click_ as it is not limited to mouse interaction. See MS Style Guide on [click](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/c/click) and on [describing UI interaction ins general](https://learn.microsoft.com/en-us/style-guide/procedures-instructions/describing-interactions-with-ui)

3. Verify that your YouTube Data API v3 was enabled and click 'Go to credentials'

> 4. Navigate to APIs & Services (left sidebar) > [Credentials](https://console.cloud.google.com/apis/credentials)

Reviewer 2: ```suggestion
4. Navigate to **APIs & Services** (left sidebar) > [Credentials](https://console.cloud.google.com/apis/credentials)
```

5. Click on the field on the left of the screen, **OAuth Consent Screen**.
6. Select **External** and **Create**.

> 7. Set the *App Name* (the name of the application asking for consent) to anything you want, e.g., *Home Assistant*.

Reviewer 2: ```suggestion
7. Set the **App Name** (the name of the application asking for consent) to anything you want, e.g., *Home Assistant*.
```

> 8. You then need to select a *Support email*. To do this, click the drop-down box and select your email address.

Reviewer 2: ```suggestion
8. You then need to select a **Support email**. To do this, from the dropdown menu, select your email address.
```

> 9. You finally need to complete the section: *Developer contact information*. To do this, enter your email address (the same as above is fine).

Reviewer 2: ```suggestion
9. You finally need to complete the section: **Developer contact information**. To do this, enter your email address (the same as above is fine).
```

10. Scroll to the bottom and click **Save and Continue**. You don't have to fill out anything else, or it may enable additional review.

> 11. You will then be automatically taken to the Scopes page. You do not need to add any scopes here, so click Save and Continue to move to the Optional info page. You do not need to add anything to the Optional info page, so click Save and Continue, which will take you to the Summary page. Click Back to Dashboard.

Reviewer 2: ```suggestion
11. You will then be automatically taken to the **Scopes** page. You do not need to add any scopes here, so click **Save and Continue** to move to the **Optional info** page. You do not need to add anything to the **Optional info** page, so select **Save and Continue**, which will take you to the **Summary** page. Select **Back to Dashboard**.
```

> 12. Click **OAuth consent screen** again and set *Publish Status* to **Production** otherwise your credentials will expire every 7 days.

Reviewer 2: ```suggestion
12. Select **OAuth consent screen** again and set *Publish Status* to **Production**. Otherwise your credentials will expire every 7 days.
```

13. Make sure **Publishing status** is set to production.

> 14. Click **Credentials** in the menu on the left-hand side of the screen, then click **Create credentials** (at the top of the screen), then select *OAuth client ID*.

Reviewer 2: ```suggestion
14. Select **Credentials** in the menu on the left-hand side of the screen, then select **Create credentials** (at the top of the screen), then select **OAuth client ID**.
```

> 15. Set the Application type to *Web application* and give this credential set a name (like "Home Assistant Credentials").

Reviewer 2: ```suggestion
15. Set the Application type to **Web application** and give this credential set a name (like "Home Assistant Credentials").
```

> 16. Add https://my.home-assistant.io/redirect/oauth to *Authorized redirect URIs* then click **Create**. This is not a placeholder and is the URI that must be used.

Reviewer 2: ```suggestion
16. Add `https://my.home-assistant.io/redirect/oauth` to **Authorized redirect URIs** then select **Create**. This is not a placeholder and is the URI that must be used.
```

> 17. You will then be presented with a pop-up saying *OAuth client created* showing *Your Client ID* and *Your Client Secret*. Make a note of these (for example, copy and paste them into a text editor), as you will need these shortly. Once you have noted these strings, click **OK**. If you need to find these credentials again at any point, then navigate to *APIs & Services > Credentials*, and you will see *Home Assistant Credentials* (or whatever you named them in the previous step) under *OAuth 2.0 Client IDs*. To view both the *Client ID* and *Client secret*, click on the pencil icon; this will take you to the settings page for these credentials, and the information will be on the right-hand side of the page.

Reviewer 2: ``` suggestion
17. You will then be presented with a pop-up saying **OAuth client created** showing **Your Client ID** and **Your Client Secret**. Make a note of these (for example, copy and paste them into a text editor), as you will need them shortly. Once you have noted these strings, click **OK**. If you need to find these credentials again at any point, then navigate to **APIs & Services** > **Credentials**, and you will see **Home Assistant Credentials** (or whatever you named them in the previous step) under **OAuth 2.0 Client IDs**. To view both the **Client ID** and **Client secret**, select the pencil icon. This will take you to the settings page for these credentials, and the information will be on the right-hand side of the page.
```

> 18. Double-check that the *YouTube Data API v3* has been automatically enabled. To do this, select **Library** from the menu, then search for *YouTube Data API v3*. If it is enabled you will see *API Enabled* with a green tick next to it. If it is not enabled, then enable it.

Reviewer 2: ```suggestion
18. Double-check that the **YouTube Data API v3** has been automatically enabled. To do this, select **Library** from the menu, then search for **YouTube Data API v3**. If it is enabled, you will see **API Enabled** with a green tick next to it. If it is not enabled, then enable it.
```
I noticed that sometimes you use italics to markup UI strings. Any particular reason? Normally, we just use bold to markup UI strings.

Author: To be honest, since the way of setting up is the same as for the google-mail integration, I just copied that over and edited the APIs that are needed. When this gets merged I will update these comments on that page aswell

{% enddetails %}

{% include integrations/config_flow.md %}

The integration setup will next give you instructions to enter the [Application Credentials](/integrations/application_credentials/) (OAuth Client ID and Client Secret) and authorize Home Assistant to connect to YouTube.

{% details "OAuth and Device Authorization steps" %}

1. Continue through the steps of selecting the account you want to authorize.

2. **NOTE**: You may get a message telling you that the app has not been verified and you will need to acknowledge that in order to proceed.

> 3. You can now see the details of what you are authorizing Home Assistant to access with two options at the bottom. Click **Continue**.

Reviewer 2: ```suggestion
3. You can now see the details of what you are authorizing Home Assistant to access with two options at the bottom. Select **Continue**.
```

> 4. The page will now display *Link account to Home Assistant?*, note *Your instance URL*. If this is not correct, please refer to [My Home Assistant](/integrations/my). If everything looks good, click **Link Account**.

Reviewer 2: ```suggestion
4. The page will now display **Link account to Home Assistant?**, note **Your instance URL**. If this is not correct, refer to [My Home Assistant](/integrations/my). If everything looks good, select **Link Account**.
```

> 5. You may close the window, and return back to Home Assistant where you should see a *Success!* message from Home Assistant.

Reviewer 2: ```suggestion
5. You may close the window, and return back to Home Assistant where you should see a **Success!** message from Home Assistant.
```
